geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/driver/KryoLazyDensityDriver.scala

```diff
@@ -0,0 +1,107 @@
+/** *********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+* ************************************************************************/
+
+package org.locationtech.geomesa.hbase.driver
+
+import java.io.IOException
+import java.util.{ArrayList, List}
+
+import com.google.protobuf.{ByteString, RpcCallback, RpcController}
+import org.apache.hadoop.hbase.client.Table
+import org.apache.hadoop.hbase.client.coprocessor.Batch._
+import org.apache.hadoop.hbase.ipc.BlockingRpcCallback
+import org.locationtech.geomesa.hbase.proto.KryoLazyDensityProto._
+
+import scala.collection.JavaConversions._
+
+/**
+  * An RpcController implementation for use here in this endpoint.
+  */
+class KryoLazyDensityRpcController extends RpcController {
+
+  var errorText: String = _
+
+  private var cancelled: Boolean = false
+
+  var failed: Boolean = false
```

Comment:
we should keep `failed` and `errorText` as private if possible, to avoid leaking mutable state. You can also use something like `private [driver]` or `private [hbase]` if you need access outside this instance.